File: fetch_beijing_tier3_hospitals.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"
	"time"
)

type POI struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	Location string `json:"location"`
	Type     string `json:"type"`
	Typecode string `json:"typecode"`
	Address  string `json:"address"`
	Tel      string `json:"tel"`
}

type APIResp struct {
	Pois []POI `json:"pois"`
}

func main() {
	key := os.Getenv("AMAP_KEY")
	if key == "" {
		fmt.Println("请先设置环境变量AMAP_KEY")
		return
	}
	city := "北京"
	keyword := "三甲医院"
	outputFile := "beijing_tier3_hospitals_by_keyword_go.json"

	allPois := make([]POI, 0)
	page := 1
	for {
		url := fmt.Sprintf("https://restapi.amap.com/v3/place/text?key=%s&keywords=%s&city=%s&citylimit=true&offset=25&page=%d", key, keyword, city, page)
		resp, err := http.Get(url)
		if err != nil {
			fmt.Println("请求失败:", err)
			break
		}
		defer resp.Body.Close()
		body, _ := ioutil.ReadAll(resp.Body)
		var data APIResp
		json.Unmarshal(body, &data)
		if len(data.Pois) == 0 {
			break
		}
		allPois = append(allPois, data.Pois...)
		fmt.Printf("已抓取第%d页，累计%d条\n", page, len(allPois))
		if len(data.Pois) < 25 {
			break
		}
		page++
		time.Sleep(500 * time.Millisecond)
	}

	// 去重
	unique := make(map[string]POI)
	for _, poi := range allPois {
		unique[poi.ID] = poi
	}
	result := make([]POI, 0, len(unique))
	for _, poi := range unique {
		result = append(result, poi)
	}
	b, _ := json.MarshalIndent(result, "", "  ")
	ioutil.WriteFile(outputFile, b, 0644)
	fmt.Printf("最终去重后共%d条，已保存到 %s\n", len(result), outputFile)
	fmt.Println("前10条POI示例:")
	for i, poi := range result {
		if i >= 10 {
			break
		}
		fmt.Printf("%d. %s | %s | %s\n", i+1, poi.Name, poi.Address, poi.ID)
	}
}

File: fetch_beijing_tier3_hospitals.py
import requests
import json
import time

AMAP_KEY = '请填写你的高德KEY'  # TODO: 替换为你的高德API KEY
CITY = '北京'
KEYWORD = '三级甲等医院'
TYPECODE = '090101'
OUTPUT = 'beijing_tier3_hospitals.json'

all_pois = []
page = 1
while True:
    url = (
        f'https://restapi.amap.com/v3/place/text?key={AMAP_KEY}'
        f'&keywords={KEYWORD}&city={CITY}&types={TYPECODE}&citylimit=true&offset=25&page={page}'
    )
    resp = requests.get(url)
    data = resp.json()
    pois = data.get('pois', [])
    if not pois:
        break
    all_pois.extend(pois)
    print(f'已抓取第{page}页，累计{len(all_pois)}条')
    if len(pois) < 25:
        break
    page += 1
    time.sleep(0.5)  # 防止QPS过高

# 去重
unique = {}
for poi in all_pois:
    unique[poi['id']] = {
        'id': poi['id'],
        'name': poi['name'],
        'location': poi['location'],
        'typecode': poi['typecode'],
        'address': poi.get('address', ''),
        'tel': poi.get('tel', ''),
    }

with open(OUTPUT, 'w', encoding='utf-8') as f:
    json.dump(list(unique.values()), f, ensure_ascii=False, indent=2)

print(f'北京三甲医院POI名单已保存到 {OUTPUT}，共{len(unique)}条') 